### PR TITLE
IBX-1784: Added getDefaultLanguageCode method to API/Repository/Values/Content/Content

### PR DESCRIPTION
**Pull request by mateuszbieniek**
Kernel side changes for https://github.com/ezsystems/ezplatform-http-cache/pull/166

### CODE REVIEW

**Comment by mateuszbieniek** [comment]
Failed SonarCloud Code Analysis tests is a false positive for me.

**Review by alongosz — CHANGES_REQUESTED** [review]
We're missing test coverage here. Integration one would be enough. ~, but let's resolve https://github.com/ezsystems/ezplatform-http-cache/pull/166#pullrequestreview-855318782 first.~

**Comment by mateuszbieniek** [comment]
CS test fail looks like a false positive.

File: eZ/Publish/API/Repository/Tests/Values/Content/DefaultLanguageCodeForContentTest.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            'eng-US' => 'Test US',
+        ];
+        $testFolder = $this->createFolder(
+            $names,
+            2,
+            null,
+            false
+        );
+
+        $this->assertEquals('eng-GB', $testFolder->getDefaultLanguageCode());
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/Values/Content/DefaultLanguageCodeForContentTest.php]: CS
```suggestion
        self::assertEquals('eng-GB', $testFolder->getDefaultLanguageCode());
```

File: eZ/Publish/API/Repository/Tests/Values/Content/DefaultLanguageCodeForContentTest.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Content;
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/Values/Content/DefaultLanguageCodeForContentTest.php]: This namespace is incorrect. Moreover new classes should be placed in the new structure, so `test/` directory.

File: eZ/Publish/API/Repository/Tests/Values/Content/DefaultLanguageCodeForContentTest.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            $names,
+            2,
+            null,
+            false
+        );
+
+        $repository = $this->getRepository();
+        $testFolderInGerman = $repository->getContentService()->loadContent($testFolder->id, ['ger-DE']);
+
+        $this->assertEquals('ger-DE', $testFolderInGerman->getDefaultLanguageCode());
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/Values/Content/DefaultLanguageCodeForContentTest.php]:
```suggestion
        self::assertEquals('ger-DE', $testFolderInGerman->getDefaultLanguageCode());
```

**Review by alongosz — CHANGES_REQUESTED** [review]
Current CS issue is not related to the changes, I'll look into it separately.
Remarks:

File: eZ/Publish/API/Repository/Tests/Values/Content/DefaultLanguageCodeForContentTest.php (hunk 2 above)
Comment (mateuszbieniek) [reply eZ/Publish/API/Repository/Tests/Values/Content/DefaultLanguageCodeForContentTest.php]: I don't know how this happened O.O
I blame PHPStorm over Jetbrains Gateway

File: tests/lib/Repository/Values/Content/DefaultLanguageCodeForContentTest.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Core\Repository\Values\Content;
```
Comment (alongosz) [inline tests/lib/Repository/Values/Content/DefaultLanguageCodeForContentTest.php]: If you look at `ibexa/core` directory/namespace structure, this namespace is reserved for unit tests. AFAICS it should be rather placed in the new directory `./tests/integration/Core/Repository/Values/Content` as:
```suggestion
namespace Ibexa\Tests\Integration\Core\Repository\Values\Content;
```

**Review by alongosz — CHANGES_REQUESTED** [review]

File: tests/lib/Repository/Values/Content/DefaultLanguageCodeForContentTest.php (hunk 4 above)
Comment (mateuszbieniek) [reply tests/lib/Repository/Values/Content/DefaultLanguageCodeForContentTest.php]: Ok - on it

**Review by alongosz — APPROVED** [review]

**Review by adamwojs — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Review by mnocon — APPROVED** [review]
Tested together with https://github.com/ezsystems/ezplatform-http-cache/pull/166

**Pull request merged**